plugin-server/src/worker/ingestion/team-manager.ts

```diff
@@ -78,7 +78,9 @@ export class TeamManager {
         const timeout = timeoutGuard(`Still running "fetchTeam". Timeout warning after 30 sec!`)
         try {
             const team: Team | null = (await this.db.fetchTeam(teamId)) || null
-            this.teamCache.set(teamId, team)
+            if (team) {
```

Comment:
Is this a good idea? This means that we do a team lookup if team has been deleted 100% of the time.